backend/danswer/connectors/confluence/connector.py

```diff
@@ -253,126 +191,87 @@ class RecursiveIndexer:
     def __init__(
         self,
         batch_size: int,
-        confluence_client: DanswerConfluence,
+        confluence_client: Confluence,
         index_recursively: bool,
         origin_page_id: str,
     ) -> None:
-        self.batch_size = 1
-        # batch_size
+        self.batch_size = batch_size
         self.confluence_client = confluence_client
         self.index_recursively = index_recursively
         self.origin_page_id = origin_page_id
-        self.pages = self.recurse_children_pages(0, self.origin_page_id)
+        self.pages = self.recurse_children_pages(self.origin_page_id)
 
     def get_origin_page(self) -> list[dict[str, Any]]:
         return [self._fetch_origin_page()]
 
-    def get_pages(self, ind: int, size: int) -> list[dict]:
-        if ind * size > len(self.pages):
-            return []
-        return self.pages[ind * size : (ind + 1) * size]
+    def get_pages(self) -> list[dict[str, Any]]:
+        return self.pages
 
-    def _fetch_origin_page(
-        self,
-    ) -> dict[str, Any]:
+    def _fetch_origin_page(self) -> dict[str, Any]:
         get_page_by_id = make_confluence_call_handle_rate_limit(
             self.confluence_client.get_page_by_id
         )
         try:
             origin_page = get_page_by_id(
-                self.origin_page_id, expand="body.storage.value,version"
+                self.origin_page_id, expand="body.storage.value,version,space"
             )
             return origin_page
         except Exception as e:
             logger.warning(
-                f"Appending orgin page with id {self.origin_page_id} failed: {e}"
+                f"Appending origin page with id {self.origin_page_id} failed: {e}"
             )
             return {}
 
     def recurse_children_pages(
         self,
-        start_ind: int,
         page_id: str,
     ) -> list[dict[str, Any]]:
         pages: list[dict[str, Any]] = []
```

Comment:
pablo did a bunch of changes that seem to make the recursive indexing stuff work